editor/groups_editor.cpp

```diff
@@ -59,776 +61,791 @@ static bool can_edit(Node *p_node, String p_group) {
+		}
+		if (!scene_groups[E]) {
+			item->add_button(0, get_theme_icon(SNAME("Lock"), SNAME("EditorIcons")), -1, true, TTR("This group belongs to another scene and can't be edited."));
 		}
+		item->add_button(0, get_theme_icon(SNAME("ActionCopy"), SNAME("EditorIcons")), COPY_GROUP, false, TTR("Copy group name to clipboard."));
 	}
 
-	renamed_group->set_text(0, name); // Spaces trimmed.
+	List<StringName> keys;
+	for (const KeyValue<StringName, String> &E : global_groups) {
+		keys.push_back(E.key);
+	}
+	keys.sort_custom<NoCaseComparator>();
 
-	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
-	undo_redo->create_action(TTR("Rename Group"));
+	TreeItem *global_root = tree->create_item(root);
+	global_root->set_text(0, "Global Groups");
+	global_root->set_icon(0, get_theme_icon(SNAME("Environment"), SNAME("EditorIcons")));
+	global_root->set_custom_bg_color(0, get_theme_color(SNAME("prop_subsection"), SNAME("Editor")));
+	global_root->set_selectable(0, false);
 
-	List<Node *> nodes;
-	scene_tree->get_nodes_in_group(selected_group, &nodes);
-	bool removed_all = true;
-	for (Node *node : nodes) {
-		if (can_edit(node, selected_group)) {
-			undo_redo->add_do_method(node, "remove_from_group", selected_group);
-			undo_redo->add_undo_method(node, "remove_from_group", name);
-			undo_redo->add_do_method(node, "add_to_group", name, true);
-			undo_redo->add_undo_method(node, "add_to_group", selected_group, true);
-		} else {
-			removed_all = false;
+	for (const StringName &E : keys) {
+		if (!filter->get_text().is_subsequence_ofn(E)) {
+			continue;
 		}
+		TreeItem *item = tree->create_item(global_root);
+		item->set_cell_mode(0, TreeItem::CELL_MODE_CHECK);
+		item->set_editable(0, true);
+		item->set_checked(0, current_groups.find(E) != nullptr);
+		item->set_text(0, E);
+		item->set_meta("__local", false);
+		item->set_meta("__name", E);
+		item->set_meta("__description", global_groups[E]);
+		if (!global_groups[E].is_empty()) {
+			item->set_tooltip_text(0, vformat("%s\n\n%s", E, global_groups[E]));
+		}
+		item->add_button(0, get_theme_icon(SNAME("ActionCopy"), SNAME("EditorIcons")), COPY_GROUP, false, TTR("Copy group name to clipboard."));
 	}
 
-	if (!removed_all) {
-		undo_redo->add_do_method(this, "_add_group", selected_group);
-		undo_redo->add_undo_method(this, "_delete_group_item", selected_group);
-	}
-
-	undo_redo->add_do_method(this, "_rename_group_item", selected_group, name);
-	undo_redo->add_undo_method(this, "_rename_group_item", name, selected_group);
-	undo_redo->add_do_method(this, "_group_selected");
-	undo_redo->add_undo_method(this, "_group_selected");
-	undo_redo->add_do_method(this, "emit_signal", "group_edited");
-	undo_redo->add_undo_method(this, "emit_signal", "group_edited");
+	updating_tree = false;
+}
 
-	undo_redo->commit_action();
+void GroupsEditor::_global_group_changed() {
+	_update_groups();
+	call_deferred("_update_tree");
```

Comment:
This can be changed to `callable_mp()` too (I mean here only; even if a method is bound, it's still better to call it directly when possible).